Jenkinsfile

```diff
@@ -395,7 +395,7 @@ lock(resource: "build-${params.STREAM}") {
             // Key off of s3_stream_dir: i.e. if we're configured to upload artifacts
             // to S3, we also take that to mean we should upload an AMI. We could
             // split this into two separate developer knobs in the future.
-            if (s3_stream_dir) {
+            if (s3_stream_dir && !(params.STREAM in streams.mechanical)) {
```

Comment:
Maybe hoist out a
`Boolean not_mechanical = !(params.STREAM in streams.mechanical);` and use it elsewhere?

Reply:
Thanks, did that (though went with `is_mechanical`).